js/components/splash-screen.js

```diff
@@ -0,0 +1,60 @@
+import React, { PureComponent } from 'react'
+import splashScreen from '../../images/splash-screen.png'
+import ccLogo from '../../images/cc-logo.png'
+
+import css from '../../css-modules/splash-screen.less'
+
+const HIDE_AFTER = 1800 // ms
+// Note that transition duration has to match value in CSS file.
+const TRANSITION_DURATION = 500 // ms
+// If window is smaller than provided dimensions, apply scaling.
+const MIN_HEIGHT = 700 // px
+const MIN_WIDTH = 500 // px
+
+export default class SplashScreen extends PureComponent {
+  constructor (props) {
+    super(props)
+    this.state = {
+      show: true,
+      fadeOut: false,
+      scale: 1
+    }
+    this.onWindowResize = this.onWindowResize.bind(this)
+  }
+
+  componentDidMount () {
+    setTimeout(() => {
+      this.setState({ fadeOut: true })
+    }, HIDE_AFTER - TRANSITION_DURATION)
+    setTimeout(() => {
+      this.setState({ show: false })
```

Comment:
Small, but I wonder if it's worth removing the event listener once the splash screen hides? Seems like it never unmounts so it just sticks around.

Reply:
Good catch, thanks.

I actually removed this listener at all. I was using it when I was playing with splash screen displayed all the time. But in reality, it's only on page load for a second or two. So, we can read window size on mount and just use it. I don't think anyone would be able to resize window while it's visible. 😉